cpp/Ice/greeter/README.md

```diff
@@ -1,14 +1,25 @@
 # Greeter
 
-The Greeter demo illustrates how to send a request and wait for the response.
+The Greeter demo illustrates how to build a simple application with Ice.
 
-To build the demo run:
+The client demonstrates the 3 APIs you can use to make remove invocations using generated Ice proxies:
+
+- synchronous (`greet`),
+- asynchronous with future (`greetAsync`); and
+- asynchronous with callbacks (another `greetAsync`)
+
+This demo alo provides two implementations for the server: a synchronous dispatch implementation (`server`), and an
+asynchronous dispatch implementation (`serveramd`). The same client works with both.
```

Comment:
I don't see why we say `same`. There's only one client in this demo right?
What is this client the "same" as?
```suggestion
asynchronous dispatch implementation (`serveramd`). The client works with both.
```